src/AzSK.ADO/Framework/Core/SVT/SVTControlAttestation.ps1

```diff
@@ -282,6 +282,18 @@ class SVTControlAttestation
 					$controlState.State.ApprovedExceptionID = $this.attestOptions.ApprovedExceptionID
 					$controlState.State.ExpiryDate = $exceptionApprovalExpiryDate.ToString("MM/dd/yyyy");
 				}
+
+				# Checking if the resource id is present in extended expiry list of control settings
+				if([Helpers]::CheckMember($this.ControlSettings, "ExtendedAttestationExpiryResources") -and [Helpers]::CheckMember($this.ControlSettings, "ExtendedAttestationExpiryDuration")){
+					if(($this.ControlSettings.ExtendedAttestationExpiryResources | Get-Member "ResourceType") -and ($this.ControlSettings.ExtendedAttestationExpiryResources | Get-Member "ResourceIds")) {
+						$extended_resources = $this.ControlSettings.ExtendedAttestationExpiryResources | Where { $_.ResourceType -match $controlItem.ResourceContext.ResourceTypeName }
+						# type null check
+						if($controlState.ResourceId -in $extended_resources.ResourceIds){
```

Comment:
This scenario will fail if you have 'ServiceConnection' in ExtendedAttestationExpiryResources, but you are scanning other resource type. please verify all scenarios.